File: spark-python/pyspark-sql/src/sql/catalog/spark_catalog.py
import os

from pyspark.sql import SparkSession

if __name__ == '__main__':
    derby_home = os.environ["derby.system.home"]
    warehouse_directory = os.environ["SPARK_WAREHOUSE"]
    spark = (SparkSession
             .builder
             .master("local[*]")
             .appName("json-sql")
             .config("spark.sql.warehouse.dir", warehouse_directory)
             .config("spark.driver.extraJavaOptions",
                     f"-Dderby.system.home='{derby_home}'")
             .enableHiveSupport()
             .getOrCreate()
             )

    # List all databases
    for database in spark.catalog.listDatabases():
        print(database.name)

    print(spark.catalog.currentDatabase())

    # listing all tables in a current database
    for table in spark.catalog.listTables():
        print(table.name)

    # set spark_hive database
    print(spark.catalog.setCurrentDatabase("spark_hive"))

    # listing all tables in a current database
    for table in spark.catalog.listTables():
        print(table.name)

File: spark-scala/spark-sql/spark-sql-catalog/src/main/scala/com/mahfooz/spark/catalog/SparkCatalog.scala
package com.mahfooz.spark.catalog

import org.apache.spark.sql.SparkSession

object SparkCatalog {

  def main(args: Array[String]): Unit = {
    val sparkWarehouse =
      sys.env.getOrElse("SPARK_WAREHOUSE", "spark-warehouse").replace("\\", "/")
    System.setProperty(
      "derby.system.home",
      sys.env.getOrElse("derby.system.home", "").replace("\\", "/")
    )

    val spark = SparkSession
      .builder()
      .master("local[*]")
      .appName("SparkManagedTable")
      .config("spark.sql.warehouse.dir", sparkWarehouse)
      .enableHiveSupport()
      .getOrCreate()

    //Accessing Catalog
    val catalog = spark.catalog

    //Querying the databases
    catalog.listDatabases().select("name").show()

    import spark.implicits._

    //Registering Dataframe with createTempView
    val df = Seq(
      (1,"malam",20),
      (2,"halam",3),
      (3,"hfatma",3)
    ).toDF("id","name","age")

    df.createTempView("employee")

    //Querying the tables
    catalog.listTables().select("name").show(false)

    //Checking is table cached or not
    println(catalog.isCached("employee"))

    df.cache()
    println(catalog.isCached("employee"))

    //Drop view
    catalog.dropTempView("employee")
    catalog.listTables().select("name").show(false)
  }

}
